core/vm/cvm_adapter.go

```diff
@@ -60,3 +61,7 @@ func (cvm *CVMAdapter) IntraBlockState() evmtypes.IntraBlockState {
 func (cvm *CVMAdapter) TxContext() evmtypes.TxContext {
 	return evmtypes.TxContext{}
 }
+
+func (cvm *CVMAdapter) FirehoseContext() *firehose.Context {
+	return cvm.Cvm.firehoseContext
+}
```

Comment:
why do we need to add this? is this used somewhere?

Reply:
`CVMAdapter` needs to implement `VMInterface` which has `FirehoseContext` function.